File: Server/Models/Address.cs
using System.ComponentModel.DataAnnotations;

namespace Server.Models;

public class Address
{
    public int Id { get; set; }
    public int? HouseNumber { get; set; }
    [StringLength(maximumLength: 50, MinimumLength = 1)]
    public string? HouseName { get; set; }
    [StringLength(maximumLength: 50, MinimumLength = 0)]
    public string? Street { get; set; }
    [StringLength(maximumLength: 50, MinimumLength = 1)]
    public string? County { get; set; }
    [StringLength(maximumLength: 50, MinimumLength = 1)]
    public string? Country { get; set; }
    [StringLength(maximumLength: 50, MinimumLength = 1)]
    public string? PostCode { get; set; }
}

File: client/lib/data/models/address.dart
import 'dart:convert';


List<Address> addressFromJson(String str) =>
  List<Address>.from(json.decode(str).map((x) => Address.fromJson(x)));

String addressToJson(List<Address> data) =>
  json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class Address{
  final int id;
  final int houseNumber;
  final String houseName;
  final String street;
  final String county;
  final String country;
  final String postCode;

  const Address({
    required this.id,
    required this.houseNumber,
    required this.houseName,
    required this.street,
    required this.county,
    required this.country,
    required this.postCode,
  });

  factory Address.fromJson(Map<String, dynamic> json) => Address(
    id: json['id'],
    houseNumber: json['houseNumber'],
    houseName: json['houseName'],
    street: json['location'],
    county: json['county'],
    country: json['country'],
    postCode: json['postCode'],
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'houseNumber': houseNumber,
    'houseName': houseName,
    'street': street,
    'county': country,
    'country': country,
    'postCode': postCode,
  };
}